`kubescape operator scan` panics with nil pointer dereference when no kubeconfig is available

# Description
Running any `kubescape operator scan ...` command without a valid kubeconfig (or with `KUBECONFIG` pointing at a missing file) does not produce a user-facing error,  it panics with a nil pointer dereference and dumps a full Go runtime stack trace to the user's terminal. The same scenario for `kubescape scan` is handled cleanly (logs an error and exits 1); the operator subcommand is the outlier.

Reproducer (no special setup, either no kubeconfig, or a kubeconfig that points at an unreachable cluster):

```
$ unset KUBECONFIG; mv ~/.kube/config{,.bak}    # simulate fresh machine
$ /tmp/kubescape operator scan configurations
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8173e67]

goroutine 1 [running]:
github.com/kubescape/kubescape/v3/core/core.getOperatorPod(0x0, {0x98cfc7c, 0x9})
    /home/.../core/core/clusterconnector.go:34 +0x87
github.com/kubescape/kubescape/v3/core/core.NewOperatorAdapter({0xa3cb630, ...}, ...)
    /home/.../core/core/clusterconnector.go:47 +0x3f
github.com/kubescape/kubescape/v3/cmd/operator.getOperatorScanConfigCmd.func2(...)
    /home/.../cmd/operator/configscan.go:32 +0x2a
...
```

Root cause in `core/core/clusterconnector.go:45-49`:

```go
func NewOperatorAdapter(scanInfo cautils.OperatorScanInfo, ns string) (*OperatorAdapter, error) {
    k8sClient := getKubernetesApi()           // returns nil when no kubeconfig
    pod, err := getOperatorPod(k8sClient, ns) // immediate use of nil
    ...
}
```

And the deref site at `clusterconnector.go:30-34`:

```go
func getOperatorPod(k8sClient *k8sinterface.KubernetesApi, ns string) (*v1.Pod, error) {
    listOptions := metav1.ListOptions{
        LabelSelector: "app=operator",
    }
    pods, err := k8sClient.KubernetesClient.CoreV1().Pods(ns).List(...)  // <-- nil deref
```

`getKubernetesApi()` (in `core/core/utils.go` / wherever it lives) returns nil on initialisation failure rather than panicking *there* — but the operator adapter trusts the return value without checking. Both `operator scan configurations` and `operator scan vulnerabilities` go through this exact path and both panic identically.

# Environment
- OS: `Ubuntu 22.04 LTS` (Linux 6.17)
- Version: `dev` (built from `main` at HEAD)
- Reproducer: any machine where `k8sinterface.IsConnectedToCluster()` returns false. This includes (a) no kubeconfig at all, (b) kubeconfig pointing at a cluster that is unreachable (DNS failure, deleted kind cluster, VPN down, etc.), and (c) running outside a pod with no kubeconfig present.

# Steps To Reproduce
```
# any of these reproduces; the precondition is "no working kube client"
$ /tmp/kubescape operator scan configurations
$ /tmp/kubescape operator scan vulnerabilities
```

# Expected behavior
A clean, user-facing error matching what other commands do in the same scenario, e.g.:

```
[fatal] Could not connect to cluster: invalid configuration — set --kubeconfig, $KUBECONFIG, or $KUBERNETES_MASTER.
```

Exit code 1, no stack trace, no usage dump.

# Actual Behavior
Reproduction matrix (no kubeconfig):

| Command | Result |
|---|---|
| `kubescape scan` (cluster scan) | `[fatal] ...could not load... invalid configuration` — exit 1, clean |
| `kubescape prerequisites` | `[fatal] Could not create kube client. Exiting.` — exit 1, clean (though message could be friendlier; see issue-7) |
| **`kubescape operator scan configurations`** | **`panic: nil pointer dereference` + Go runtime stack trace** — exit 2 |
| **`kubescape operator scan vulnerabilities`** | **Same panic** |
| `kubescape vap deploy-library` | works (doesn't need cluster) |

This is a regression-worthy UX: any user who tries the operator subcommand without already having an operator installed will see a crash and reasonably assume the binary is broken.

# Suggested fix
Two-line guard in `NewOperatorAdapter` plus a tightening in `getOperatorPod`:

```go
// core/core/clusterconnector.go:45
func NewOperatorAdapter(scanInfo cautils.OperatorScanInfo, ns string) (*OperatorAdapter, error) {
    k8sClient := getKubernetesApi()
    if k8sClient == nil || k8sClient.KubernetesClient == nil {
        return nil, fmt.Errorf("could not connect to cluster: set --kubeconfig, $KUBECONFIG, or $KUBERNETES_MASTER")
    }
    pod, err := getOperatorPod(k8sClient, ns)
    ...
}

// core/core/clusterconnector.go:30 — defensive, in case any other caller path slips through
func getOperatorPod(k8sClient *k8sinterface.KubernetesApi, ns string) (*v1.Pod, error) {
    if k8sClient == nil || k8sClient.KubernetesClient == nil {
        return nil, errors.New("kubernetes client is not initialised")
    }
    ...
}
```

Better still, make `getKubernetesApi()` return `(*k8sinterface.KubernetesApi, error)` instead of swallowing the error - every caller that does `client := getKubernetesApi()` and then uses `client.KubernetesClient` is a latent panic. A grep for `getKubernetesApi()` callers will show how many sites need this guard.

# Source
- `core/core/clusterconnector.go:30-43` — `getOperatorPod` deref of nil `KubernetesClient`
- `core/core/clusterconnector.go:45-61` — `NewOperatorAdapter` calls without checking
- `cmd/operator/configscan.go:32` — invocation site (configurations)
- `cmd/operator/vulnerabilitiesscan.go:33` — invocation site (vulnerabilities), same pattern
- `core/core/initutils.go:24-29` — `getKubernetesApi()` returns nil whenever `k8sinterface.IsConnectedToCluster()` is false (no kubeconfig OR kubeconfig present but cluster unreachable)
- Any other call site of `getKubernetesApi()` that doesn't nil-check is exposed to the same class of panic

--- a/core/core/clusterconnector.go
+++ b/core/core/clusterconnector.go
@@ -28,6 +28,9 @@ type OperatorAdapter struct {
 }
 
 func getOperatorPod(k8sClient *k8sinterface.KubernetesApi, ns string) (*v1.Pod, error) {
+	if k8sClient == nil || k8sClient.KubernetesClient == nil {
+		return nil, errors.New("kubernetes client is not initialised")
+	}
 	listOptions := metav1.ListOptions{
 		LabelSelector: "app=operator",
 	}
@@ -44,6 +47,9 @@ func getOperatorPod(k8sClient *k8sinterface.KubernetesApi, ns string) (*v1.Pod, 
 
 func NewOperatorAdapter(scanInfo cautils.OperatorScanInfo, ns string) (*OperatorAdapter, error) {
 	k8sClient := getKubernetesApi()
+	if k8sClient == nil || k8sClient.KubernetesClient == nil {
+		return nil, errors.New("could not connect to cluster: set --kubeconfig, $KUBECONFIG, or $KUBERNETES_MASTER")
+	}
 	pod, err := getOperatorPod(k8sClient, ns)
 	if err != nil {
 		return nil, err

--- a/core/core/clusterconnector_test.go
+++ b/core/core/clusterconnector_test.go
@@ -81,3 +81,11 @@ func Test_getOperatorPod(t *testing.T) {
 		})
 	}
 }
+
+func Test_getOperatorPod_nilClient(t *testing.T) {
+	_, err := getOperatorPod(nil, kubescapeNamespace)
+	assert.EqualError(t, err, "kubernetes client is not initialised")
+
+	_, err = getOperatorPod(&k8sinterface.KubernetesApi{}, kubescapeNamespace)
+	assert.EqualError(t, err, "kubernetes client is not initialised")
+}

